### ISSUE TITLE
Wrong MMLV levels saved with vertical/horizontal models

### ISSUE THREAD

**Issue by schrum2** [schrum2/MM-NEAT#535]
In the interactive MegaMan interface, I don't think the levels being saved in MMLV format actually match the levels I'm seeing in the interactive interface.

**Comment by schrum2** [comment]
I still don't think this is fixed. It seems to work sometimes but not always ... though it is hard to evaluate because many levels are very difficult to explore.

In any case, I think the problem is that you are generating the levels with different method calls in different places, but the MMLV levels are always generated by a call to `levelListRepresentation` which calls `MegaManGANUtil.generateOneLevelListRepresentationFromGAN`.

You need to have one method that takes a `double[] latentVector` are returns a `List<List<Integer>>` level. The `generateLevelFromGAN` method sort of does this, but uses a global variable (why?) and is not used by `playLevel` method or the `levelListRepresentation` method. However, I notice that it is used in `saveLevel`, which I admit confuses me a bit.

Basically, this code needs to be refactored, cleaned up, organized, and COMMENTED! Especially JavaDoc. Try to make sure that there is one central pathway for generating levels, no mater what approach (number of models) is used.

**Comment by cappsb** [comment]
I'll resolve those. However, if you want to check the layout of what you're saving you can choose to build the level you saved. It allows you to view the whole level without having to traverse it. When I test it out, it only appears to not work when actually playing the level, but when I go to the builder it is in fact the same. The main reason for this is the fast vertical scrolling.

**Comment by cappsb** [comment]
I have refactored a lot of code. Now the levelListRepresentation is the sole generator of all levels using any method (H, V, H+V one gan, H+V two gans). The global level variable is no longer used.

**Comment by schrum2** [comment]
I think this works now, but we'll obviously want to keep an eye on it.
